online-rest-api/src/main/java/eu/itesla_project/online/rest/api/util/OnlineDBUtils.java

```diff
@@ -211,15 +224,100 @@ public WorkflowResult getWorkflowResult(String processId, String workflowId) thr
         return null;
     }
 
+    @Override
+    public ProcessSynthesis getSynthesis(String processId) throws Exception {
+        Objects.requireNonNull(processId);
+        OnlineProcess p = null;
+        try (OnlineDb onlinedb = fact.create()) {
+            p = onlinedb.getProcess(processId);
+            if(p != null)
+            {
+                ProcessSynthesis result = new ProcessSynthesis(processId);
+                Map<Integer, StateSynthesis> statesMap = new HashMap<Integer, StateSynthesis>();
+    
+                p.getWorkflowsMap().forEach((basecase, workflowId) -> {
+                    DateTime dateTime = fmt.parseDateTime(basecase);
+                    Map<Integer, List<LimitViolation>> previolsMap = onlinedb.getViolations(workflowId, OnlineStep.LOAD_FLOW);
+    
+                    if(previolsMap != null)
+                    {
+                        previolsMap.forEach( (state, limitList) -> {               
+                            StateSynthesis statesynt = statesMap.get(state);
+                            if (statesynt == null) {
+                                statesynt = new StateSynthesis(state);
+                                statesMap.put(state, statesynt);
+                            }                  
+                            fillViolationSynthesis(dateTime, statesynt.getPreContingencyViolations(), limitList);
+                        });
+                    }
+    
+                    Map<Integer, Map<String, List<LimitViolation>>> violsMap = onlinedb
+                            .getPostContingencyViolations(workflowId);
+    
+                    if (violsMap != null) {
+                        violsMap.forEach((state, contingencyViolationMap) -> {
+                            StateSynthesis statesynt = statesMap.get(state);
+                            if (statesynt == null) {
+                                statesynt = new StateSynthesis(state);
+                                statesMap.put(state, statesynt);
+                            }
+    
+                            Map<String, List<ViolationSynthesis>> contingencyMap = statesynt.getPostContingencyViolations();
+    
+                            contingencyViolationMap.forEach( (cont, limitList) -> {
+                                List<ViolationSynthesis> violationList = contingencyMap.get(cont);
+                                if (violationList == null) {
+                                    violationList = new ArrayList();
+                                    contingencyMap.put(cont, violationList);
+                                }
+                                fillViolationSynthesis(dateTime, violationList, limitList);
+                            });
+                        });
+                    }
+                });
+    
+                result.addStateSynthesis(statesMap.values().stream().collect(Collectors.toList()));
```

Comment:
I think this could be more readable:
result.addStateSynthesis(new ArrayList<>( statesMap.values());